NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -53,21 +68,28 @@ function fuse_blocklengths(x::LabelledInteger, y::LabelledInteger)
   return labelled(unlabel(x) * unlabel(y), fuse_labels(label(x), label(y)))
 end
 
+flatten_maybe_nested(v::Vector{<:Integer}) = v
+flatten_maybe_nested(v::Vector{<:AbstractGradedUnitRange}) = reduce(vcat, blocklengths.(v))
+
 using BlockArrays: blockedrange, blocks
 function tensor_product(a1::AbstractBlockedUnitRange, a2::AbstractBlockedUnitRange)
-  blocklengths = map(vec(collect(Iterators.product(blocks(a1), blocks(a2))))) do x
-    return mapreduce(length, fuse_blocklengths, x)
-  end
+  maybe_nested = map(
+    it -> mapreduce(length, fuse_blocklengths, it),
+    Iterators.flatten((Iterators.product(blocks(a1), blocks(a2)),)),
+  )
```

Comment:
```suggestion
  maybe_nested = map(
    Iterators.flatten((Iterators.product(blocks(a1), blocks(a2)),)),
  ) do it
    return mapreduce(length, fuse_blocklengths, it)
  end
```
@ogauthe not sure if the formatting is correct but hopefully you get the idea. I'm just suggesting changing over to using the [do-block syntax](https://docs.julialang.org/en/v1/manual/functions/#Do-Block-Syntax-for-Function-Arguments), I think it looks cleaner when passing more complicated anonymous functions to higher order functions like `map`.

Reply:
Ok, although I personally find do-block harder to understand.